### PR TITLE
Support moving pieces

### PR DESCRIPTION
**Pull request by SethTeck**
This is the start to the support moving pieces card. Not yet complete.

### CODE REVIEW

**Comment by radiofreemattd** [comment]
This is looking pretty good. One thing you can do to shrink the commit if you want is to delete any files that you have not personally added code to. So, for instance, we don't need the `*.scss` file...there are a few others that you can take out, which will your total commit smaller and the changes will be easier to see.

Looking really nice, guys! :star:

**Comment by radiofreemattd** [comment]
And I know you said this is all WIP, so... :tongue:

File: app/helpers/games_helper.rb [hunk 1]
```diff
@@ -12,8 +12,13 @@ def draw_square(x_axis, y_axis)
...
     current_piece = @chess_pieces.first
+
     return unless current_piece && current_piece.position_x == x_axis && current_piece.position_y == y_axis
+
     @chess_pieces.shift
-    image_tag "#{current_piece.color.downcase}_#{current_piece.type.downcase}.png"
+
+    link_to image_tag("#{current_piece.color.downcase}_#{current_piece.type.downcase}.png",
+                      id: current_piece.id),
+            piece_path(current_piece)
```
Comment (mscoutermarsh) [inline app/helpers/games_helper.rb]: looks like this line should be pulled up to 21.

File: app/helpers/pieces_helper.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    else
+      piece_name = 'empty_square'
+    end
+
+    link_to image_tag("#{piece_name}.png", id: current_piece.id, class: piece_class),
+            piece_path(id: @selected_piece, position_x: x_axis, position_y: y_axis), method: 'PUT'
+  end
+
+  def set_background_color(x_axis, y_axis)
+    x_axis.odd? && y_axis.odd? || x_axis.even? && y_axis.even?
```
Comment (mscoutermarsh) [inline app/helpers/pieces_helper.rb]: I'd move the `lite` and `dark` into this method.

```Ruby
def set_background_color(x_axis, y_axis)
  return 'light' if (x_axis.odd? && y_axis.odd? || x_axis.even? && y_axis.even?)
  'dark'
end
```

File: app/views/pieces/show.html.slim [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    .chessboard
+      / A chessboard consists of eight (8) rows, starting (from the top)
+      / with a lite-colored row. We can create eight (8) rows by creating
+      / four (4) sets of lite-dark alternating rows
+      - 1.upto(8) do |y_axis|
+        / Create a row starting with the lite color, which we can do by
+        / creating four (4) pairs of alternating squares, starting with lite
+        .chessboard__row
+          - 1.upto(8) do |x_axis|
+            = draw_square_background(x_axis, y_axis)
```
Comment (mscoutermarsh) [inline app/views/pieces/show.html.slim]: nice!

File: spec/features/user_moves_piece_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+RSpec.feature 'User moves piece', type: :feature do
+  scenario 'successfully' do
+    game = create(:game)
+    visit game_path(game)
+
+    pawn = game.chess_pieces.find_by(position_x: 1, position_y: 2)
+    click_link('', href: piece_path(pawn))
+
+    expect(current_path).to eq piece_path(pawn)
```
Comment (mscoutermarsh) [inline spec/features/user_moves_piece_spec.rb]: :+1:

File: spec/features/user_visits_gamepages_spec.rb [hunk 5]
```diff
@@ -7,4 +7,12 @@
 
     within('.chessboard') { expect(page).to have_css 'img', count: 32 }
   end
+
+  scenario 'has links for all chess pieces' do
+    game = create(:game)
+    visit game_path(game)
+    game.chess_pieces.each do |piece|
+      expect(page).to have_link('', href: piece_path(piece))
+    end
```
Comment (mscoutermarsh) [inline spec/features/user_visits_gamepages_spec.rb]: :+1:

**Pull request merged**